### PR TITLE
Added navigation to the project

### PR DESCRIPTION
**Pull request by hernan-clich**
## Summary

Added navigation to each of the books listed on the `FlatList` component.

#### Galaxy S (or similar)

navigation_galaxyS

#### Pixel 2/3 (or similar)

navigation_pixel3

#### Pixel 3XL (or similar)

navigation_pixel3xl

## Trello Card

https://trello.com/c/NWPNbq5R/13-n-71-introducci%C3%B3n-a-react-navigation

### CODE REVIEW

File: wbooks/src/app/components/SingleBook/index.tsx [hunk 1]
```diff
@@ -8,11 +8,12 @@ interface Props {
   bookCover: string | null;
   bookTitle: string;
   bookAuthor: string;
+  onPressProp: () => void;
```
Comment (sfernandez11) [inline wbooks/src/app/components/SingleBook/index.tsx]: Why `onPressProps`? Maybe `onPressBook`?

File: wbooks/src/app/screens/BookDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { View } from 'react-native';
+import ScreenWrapper from '@components/ScreenWrapper';
+
+function BookDetails() {
+  return (
+    <ScreenWrapper>
+      <View />
+    </ScreenWrapper>
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookDetails/index.tsx]: No, this is not ok. You only need the `SafeAreaView` here, not the `StatusBar`. The RN App only need one `StatusBar` per App

File: wbooks/src/app/screens/Library/index.tsx [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
+}
+
+function Library({ navigation }: Props) {
   const renderBooks: ListRenderItem<Book> = ({ item }) => (
-    <SingleBook bookCover={item.imageUrl} bookTitle={trimLineBreaks(item.title)} bookAuthor={item.author} />
+    <SingleBook
+      bookCover={item.imageUrl}
+      bookTitle={trimLineBreaks(item.title)}
+      bookAuthor={item.author}
+      onPressProp={() => navigation.navigate(SCREENS.BOOK_DETAILS)}
```
Comment (sfernandez11) [inline wbooks/src/app/screens/Library/index.tsx]: `onPressBook`

File: wbooks/src/app/screens/Library/index.tsx [hunk 4]
```diff
@@ -1,25 +1,40 @@
...
   const keyExtractor = ({ id }: Book) => String(id);
 
   return (
-    <FlatList
-      style={styles.container}
-      data={BOOKS_MOCK}
-      renderItem={renderBooks}
-      keyExtractor={keyExtractor}
-    />
+    <ScreenWrapper>
```
Comment (sfernandez11) [inline wbooks/src/app/screens/Library/index.tsx]: Same here, only the SafeArea. It's not necessary a wrapper or new component for this

File: wbooks/src/app/components/ScreenWrapper/index.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { SafeAreaView, StatusBar } from 'react-native';
+
+import styles from './styles';
+
+interface Props {
+  children: React.ReactNode;
+}
+
+function ScreenWrapper({ children }: Props) {
```
Comment (sfernandez11) [inline wbooks/src/app/components/ScreenWrapper/index.tsx]: Delete tthis component, it's not necessary. Move the `StatusBar` to `src/app/index` and the `SafeAreaView` only in the `Library` screen

File: wbooks/src/app/index.tsx [hunk 6]
```diff
@@ -8,20 +8,29 @@
...
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName={SCREENS.LIBRARY}>
+        <Stack.Screen name={SCREENS.LIBRARY} component={Library} />
+        <Stack.Screen
+          name={SCREENS.BOOK_DETAILS}
+          component={BookDetails}
+          options={{ title: 'Book Detail' }}
+        />
+      </Stack.Navigator>
+    </NavigationContainer>
```
Comment (sfernandez11) [inline wbooks/src/app/index.tsx]: Create a new component for the `AppNavigator`:
```
  return (
    <NavigationContainer>
      <StatusBar barStyle="dark-content" />
      <AppNavigator />
    </NavigationContainer>
  );
```

And then, `src/app/components/AppNavigator`:
```
function AppNavigator() {
  return (
    <Stack.Navigator initialRouteName={SCREENS.LIBRARY}>
      <Stack.Screen name={SCREENS.LIBRARY} component={Library} />
      <Stack.Screen
        name={SCREENS.BOOK_DETAILS}
        component={BookDetails}
        options={{ title: 'Book Detail' }}
      />
    </Stack.Navigator>
  );
}
```

File: wbooks/src/app/index.tsx [hunk 7]
```diff
@@ -8,20 +8,29 @@
...
   return (
-    <>
-      <StatusBar barStyle="dark-content" />
-      <SafeAreaView style={styles.safeAreaView}>
-        <Library />
-      </SafeAreaView>
-    </>
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName={SCREENS.LIBRARY}>
+        <Stack.Screen name={SCREENS.LIBRARY} component={Library} />
```
Comment (sfernandez11) [inline wbooks/src/app/index.tsx]: Missing `title`

**Review by sfernandez11 — CHANGES_REQUESTED** [review]
Great work, some comments...

File: wbooks/src/app/components/SingleBook/index.tsx (hunk 1 above)
Comment (hernan-clich) [reply wbooks/src/app/components/SingleBook/index.tsx]: Refactored

File: wbooks/src/app/screens/BookDetails/index.tsx (hunk 2 above)
Comment (hernan-clich) [reply wbooks/src/app/screens/BookDetails/index.tsx]: Good to know, refactored

File: wbooks/src/app/screens/Library/index.tsx (hunk 4 above)
Comment (hernan-clich) [reply wbooks/src/app/screens/Library/index.tsx]: Refactored

File: wbooks/src/app/components/ScreenWrapper/index.tsx (hunk 5 above)
Comment (hernan-clich) [reply wbooks/src/app/components/ScreenWrapper/index.tsx]: Refactored.

File: wbooks/src/app/index.tsx (hunk 6 above)
Comment (hernan-clich) [reply wbooks/src/app/index.tsx]: Refactored

File: wbooks/src/app/screens/Library/index.tsx (hunk 3 above)
Comment (hernan-clich) [reply wbooks/src/app/screens/Library/index.tsx]: Refactored

File: wbooks/src/app/index.tsx (hunk 7 above)
Comment (hernan-clich) [reply wbooks/src/app/index.tsx]: Refactored

**Review by sfernandez11 — APPROVED** [review]
🥇

**Pull request merged**